src/3p-frame.js

```diff
@@ -203,7 +202,7 @@ export function addDataAndJsonAttributes_(element, attributes) {
  */
 export function getBootstrapUrl(type) {
   const extension = IS_ESM ? '.mjs' : '.js';
-  if (getMode().localDev || getMode().test) {
+  if (mode.isLocalDev() || mode.isTest()) {
```

Comment:
Can this be `!mode.isProd()`? It looks like `isLocalDev()`, `isTest()`, and `isProd()` all return `false` because the prior two still read AMP properties from the window.

Reply:
I think the window properties are read so certain testing systems (Karma/Percy?) can set a flag in the window/config even if they're working with a production-compiled binary. Tbh the "modes" are very confusing imo and I would guess a lot of code as written includes several statements like this because it may have been unclear to the authors which was "right"

Reply:
Looking at the code, `isLocalDev` includes `isTest` so the `isTest` check shouldn't be necessary here

Reply:
Good catch on `isTest`, thanks!

+1 modes are confusing. This is why I'm not sure if changing `mode.isLocalDev()` -> `!mode.isProd()` is safe to do here, but that is what the Bento document would need, since the `IS_PROD` constant is reliably written at build time and the other checks on the window will likely not be available.

Reply:
FWIW in the case where `isProd()` is true, the rest of the method gets DCE'd. All calls to these functions are replaced with boolean literals (and further DCE'd) in production binaries